docs/architecture/adr-011-reports-module.md

```diff
@@ -0,0 +1,307 @@
+  rpc RemoveReason(MsgRemoveReason) returns (MsgRemoveReasonResponse);
+}
+
+// MsgCreateReport represents the message to be used to create a report
+message MsgCreateReport {
+  // Id of the reason this report has been created for
+  required uint32 reason_id = 1;
+
+  // Message attached to this report
+  optional string message = 2;
+
+  // Address of the reporter 
+  required string reporter = 3;
+
+  // Target of the report
+  oneof Target {
+    UserData user_data = 4;
+    PostData post_data = 5;
+  }
+}
+
+// MsgCreateReportResponse represents the Msg/CreateReport response type
+message MsgCreateReportResponse {
+  // Id of the newly created report
+  required uint64 report_id = 1;
+}
+
+// MsgDeleteReport represents the message to be used when deleting a report
+message MsgDeleteReport {
+  // Id of the subspace that contains the report to be deleted 
+  required uint64 subspace_id = 1;
+  
+  // Id of the report to be deleted 
+  required uint64 report_id = 2; 
+  
+  // Address of the user deleting the report 
+  required string signer = 3;
+}
+
+// MsgDeleteReportResponse represents the Msg/DeleteReport response type
+message MsgDeleteReportResponse {}
+
+// MsgSupportReason represents the message to be used when wanting to support a reason from the module params
+message MsgSupportReason {
+  // Id of the subspace for which to support the reason
+  required uint64 subspace_id = 1;
+  
+  // Id of the reason that should be supported
+  required uint32 reason_id = 2;
+  
+  // Address of the user signing the message
+  required string signer = 3;
+}
+
+// MsgSupportReasonResponse represents the Msg/SupportReason response type
+message MsgSupportReasonResponse {}
+
+// MsgAddReason represents the message to be used when adding a new supported reason
+message MsgAddReason {
+  // Id of the subspace for which to add the reason 
+  required uint64 subspace_id = 1;
+  
+  // Title of the reason
+  required string title = 2;
+
+  // Extended description of the reason and the cases it applies to
+  optional string description = 3;
+  
+  // Address of the user adding the supported reason
+  required string signer = 4;
+}
+
+// MsgAddReasonResponse represents the Msg/AddReason response type
+message MsgAddReasonResponse {
+  // Id of the newly supported reason
+  required uint32 reason_id = 1;
+}
+
+// MsgRemoveReason represents the message to be used when removing an exiting reporting reason
+message MsgRemoveReason {
+  // Id of the subspace from which to remove the reason 
+  required uint64 subspace_id = 1;
+
+  // Id of the reason to be deleted
+  required uint32 reason_id = 2;
+
+  // Address of the user adding the supported reason
+  required string signer = 3;
+}
+
+// MsgRemoveReasonResponse represents the Msg/RemoveReason response type
+message MsgRemoveReasonResponse {
+  // Id of the newly supported reason
+  required uint32 reason_id = 1;
+}
+
+```
+
+### `Query` Service
+```protobuf
+service Query {
```

Comment:
It would be useful to have `ReportTallyResult` query to show the tally result. Moderator group can use the result to govern their subspaces.

Reply:
@dadamu Honestly I don't think that's necessary. Usually _tally results_ are related to something that is programmed to be closed in the future (eg. governance proposal voting period, poll voting period, etc). Talking about reports, there is not going to be a predefined time slot during which the reporting will be valid. For this reason, in my opinion is much better to simply allow to query the various reports for something rather than having a tally result endpoint which would return the same values anyway.

Reply:
Yeah, you are right. 

Thinking about how to propose a proposal to delete a post whose harassment report amount is greater than 100 to a subspace DAO. Since this action is done by a member of DAO with the report tally result, and it is not necessary to have a tally endpoint for it.